web/permissions/permissions.go

```diff
@@ -129,10 +138,86 @@ func listPermissions(c echo.Context) error {
 	return json.NewEncoder(resp).Encode(doc)
 }
 
+func patchPermission(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+	current, err := getPermission(c)
+	if err != nil {
+		return err
+	}
+
+	var patch permissions.Permission
+	if _, err = jsonapi.Bind(c.Request(), &patch); err != nil {
+		return err
+	}
+
+	patchSet := patch.Permissions != nil && len(*patch.Permissions) > 0
```

Comment:
Permissions is not a slice, but a permissions.Set. When testing custom marshalling/unmarshalling of this field (as a JSON.object[rule tile] -> rule), i thought I needed it to be a pointer, tourns out I was wrong, fix below.